v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/templates/SpannerChangeStreamsToElasticsearch.java

```diff
@@ -0,0 +1,175 @@
+        break;
+      }
+    }
+    if (!hasUseRunnerV2) {
+      experiments.add(USE_RUNNER_V2_EXPERIMENT);
+    }
+    options.setExperiments(experiments);
+
+    String metadataTableName =
+        options.getSpannerMetadataTableName() == null
+            ? null
+            : options.getSpannerMetadataTableName();
+
+    final RpcPriority rpcPriority = options.getRpcPriority();
+    pipeline
+        .apply(
+            SpannerIO.readChangeStream()
+                .withSpannerConfig(
+                    SpannerConfig.create()
+                        .withHost(ValueProvider.StaticValueProvider.of(options.getSpannerHost()))
+                        .withProjectId(projectId)
+                        .withInstanceId(instanceId)
+                        .withDatabaseId(databaseId))
+                .withMetadataInstance(metadataInstanceId)
+                .withMetadataDatabase(metadataDatabaseId)
+                .withChangeStreamName(changeStreamName)
+                .withInclusiveStartAt(startTimestamp)
+                .withInclusiveEndAt(endTimestamp)
+                .withRpcPriority(rpcPriority)
+                .withMetadataTable(metadataTableName))
+        .apply("DataChangeRecord To JSON", ParDo.of(new DataChangeRecordToJsonFn()))
+        .apply(
+            "WriteToElasticsearch",
+            WriteToElasticsearch.newBuilder()
+                .setOptions(options.as(SpannerChangeStreamsToElasticsearchOptions.class))
+                .build());
+
+    return pipeline.run();
+  }
+
+  static class DataChangeRecordToJsonFn extends DoFn<DataChangeRecord, String> {
+
+    @ProcessElement
+    public void process(@Element DataChangeRecord element, OutputReceiver<String> output) {
+      List<ColumnType> cols = element.getRowType();
+      ModType modType = element.getModType();
+      String tableName = element.getTableName();
+      element
+          .getMods()
+          .forEach(
+              mod -> {
+                JSONObject keysJsonExtended = new JSONObject(mod.getKeysJson());
+                keysJsonExtended.put("TableName", tableName);
+                if (modType == ModType.INSERT || modType == ModType.UPDATE) {
+                  JSONObject newValuesJson = new JSONObject(mod.getNewValuesJson());
+                  // add any properties that are present in newValuesJson to the set of keys to make
+                  // a partial (or complete) row reflecting the update (or insert)
+                  cols.forEach(
+                      col -> {
+                        if (newValuesJson.has(col.getName())) {
+                          keysJsonExtended.put(col.getName(), newValuesJson.get(col.getName()));
+                        }
+                      });
+                } else {
+                  // For DELETE mod, need the keys and a property indicating it is a delete
+                  keysJsonExtended.put("IsDelete", true);
+                }
+
+                output.output(keysJsonExtended.toString());
```

Comment:
Is the goal to output one JSON string per each data change record mod?

Would you be able to reuse this transformation? 

https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/42594e181db9a428246b2372924f9c4510e8527a/v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/spannerchangestreamstobigquery/SpannerChangeStreamsToBigQuery.java#L367